src/parser/chunkparser.hpp

```diff
@@ -0,0 +1,94 @@
+#pragma once
+#include <nlohmann/json.hpp>
+#include <array>
+
+namespace docgen {
+namespace details {
+
+// These routines will parse the chunk of text starting at begin
+// until it reaches end and update the json object.
+// When a state changes or the chunk has been fully read, the routine will finish.
+inline void read_routine(const char *&begin, const char *end, nlohmann::json& json)
+{
+    // TODO: implement parsing 
+}
+
+inline void slash_routine(const char *&begin, const char *end, nlohmann::json& json)
```

Comment:
Why do we need a "slash" routine?

Reply:
Based on our diagram, at the `slash` state, if we see either `/` or `*` then we change states to "ignore whitespace" but otherwise we go back to `read` state.

Reply:
Based on your comment in slack, I'm fine with placing this logic inside `read_routine`.